### PR TITLE
Fix includes and std for module build and add MSVC support

### PR DESCRIPTION
**Pull request by HKunogi**
With this commit I could successfully build it as a module on the editor for windows with C# support using latest godot engine 4.4 source code by simply cloning the repository under the modules folder.
image
image

### CODE REVIEW

File: register_types.cpp [hunk 1]
```diff
@@ -4,7 +4,11 @@
 #include "core/hex_mesh.h"
 #include "core/mesh.h"
 #include "core/pent_mesh.h"
+#if SOTA_MODULE
+#include "core/object/class_db.h"
```
Comment (ardazishvili) [inline register_types.cpp:8]: Thank you for fix. Could you please move this include in `src/tal/godot_core.h` and include that file instead? There are includes for "SOTA_MODULE" in the bottom of file. I would like to encapsulate all things related to difference between gdextension and module in TAL. Also, see [doc](https://sota.readthedocs.io/en/latest/content/arch/tal.html)

File: src/prism_impl/prism_hex_mesh.cpp [hunk 2]
```diff
@@ -1,6 +1,10 @@
 #include "prism_impl/prism_hex_mesh.h"
 
+#if SOTA_MODULE
+#include "core/os/memory.h"
```
Comment (ardazishvili) [inline src/prism_impl/prism_hex_mesh.cpp:4]: Ditto, I would appreciate if you update and use`src/tal/godot_core.h`

**Review by ardazishvili — COMMENTED** [review]
Thank you.

There is a nit related to encapsulation of headers for gdextension/module. Also
1. If you have configured project for MSVC could you please push it too?
2. You are welcome to update AUTHORS.md

**Comment by HKunogi** [comment]
> Thank you.
> 
> There is a nit related to encapsulation of headers for gdextension/module. Also
> 
> 1. If you have configured project for MSVC could you please push it too?
> 2. You are welcome to update AUTHORS.md

I'm building using MSVC directly from command-line, that is usually the default when calling scons from windows power shell (to use clang you would need to manually specify it). So i'm not using actual visual studio. After applying those changes on the includes and the SCSub the project builds seamlessly without any additional configuration needed.
Built successfully after moving the includes to TAL.
image

**Review by ardazishvili — APPROVED** [review]
LGTM

**Comment by ardazishvili** [comment]
> > Thank you.
> > There is a nit related to encapsulation of headers for gdextension/module. Also
> > 
> > 1. If you have configured project for MSVC could you please push it too?
> > 2. You are welcome to update AUTHORS.md
> 
> I'm building using MSVC directly from command-line, that is usually the default when calling scons from windows power shell (to use clang you would need to manually specify it). So i'm not using actual visual studio. After applying those changes on the includes and the SCSub the project builds seamlessly without any additional configuration needed. Built successfully after moving the includes to TAL. image

Got it, thanks for update

**Comment by yusi-wr** [comment]
Awesome

**Pull request merged**